src/integrated/pond_soil_model.jl

```diff
@@ -193,27 +193,29 @@ ensuring the infiltration used for the boundary condition of soil
 is also used to compute the runoff for the surface water.
 
 """
-struct PrognosticRunoff{FT} <: Pond.AbstractSurfaceRunoff{FT}
-    precip::Function
+struct PrognosticRunoff{F <: Function} <: Pond.AbstractSurfaceRunoff
+    precip::F
 end
 
 """
     function Pond.surface_runoff(
-        runoff::PrognosticRunoff{FT},
+        runoff::PrognosticRunoff,
         Y::ClimaCore.Fields.FieldVector,
         p::NamedTuple,
         t,
-    ) where {FT}
+    )
 
-Extension of the `Pond.surface_runoff` function, which computes the surface runoff, for use in an LSM when the runoff is determined
+Extension of the `Pond.surface_runoff` function, which computes
+ the surface runoff, for use in an LSM when the runoff is determined
 prognostically.
 """
 function Pond.surface_runoff(
-    runoff::PrognosticRunoff{FT},
+    runoff::PrognosticRunoff,
     Y::ClimaCore.Fields.FieldVector,
     p::NamedTuple,
     t,
-) where {FT}
+)
+    FT = FTfromY(Y)
```

Comment:
why is it preferred to get `FT` like this?

Reply:
Before, we were getting `FT` from the type. However, the type does no longer encode `FT`. So, we have to get it from somewhere else. The main source of truth for that is the `space` (`Field`s have to share the same floating point type as the `space), so ideally we would get it with `Spaces.undertype(Y)`. Now, this does not work because `Y` is a `FieldVector`, so for the moment I am getting it as with `eltype`.